### PR TITLE
Add Aliyun OSS image loader

### PR DESCRIPTION
**Pull request by Gerhut**
I'm working on a next.js project behind [Aliyun (Alibaba Cloud) OSS](https://www.alibabacloud.com/help/doc-detail/31817.htm), and wrote an image loader based on [Aliyun OSS image processing](https://www.alibabacloud.com/help/doc-detail/183902.htm).

Contribute back to see if it is needed by others.

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| buildDuration | 12.8s | 13s | ⚠️ +152ms |
| nodeModulesSize | 84.9 MB | 84.9 MB | ⚠️ +1.51 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.386 | 2.364 | -0.02 |
| / avg req/sec | 1048 | 1057.33 | +9.33 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.194 | 1.236 | ⚠️ +0.04 |
| /error-in-render avg req/sec | 2093.37 | 2021.92 | ⚠️ -71.45 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..7d3b.js gzip | 11.3 kB | 11.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-56c34a4..e157.js gzip | 7.4 kB | 7.4 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.4 kB | 58.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 7.04 kB | 7.04 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-c8c3589..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53.1 kB | 53.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-fca3d..2eb1.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-8a9d104..0e42.js gzip | 1.55 kB | 1.55 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.96 kB | 7.96 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-d1cfb87..dule.js gzip | 1.49 kB | 1.49 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.59 kB | 5.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 995 B | 995 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| buildDuration | 14.4s | 14.1s | -291ms |
| nodeModulesSize | 84.9 MB | 84.9 MB | ⚠️ +1.51 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..7d3b.js gzip | 11.3 kB | 11.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-56c34a4..e157.js gzip | 7.4 kB | 7.4 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.4 kB | 58.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 7.04 kB | 7.04 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-c8c3589..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53.1 kB | 53.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-fca3d..2eb1.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-8a9d104..0e42.js gzip | 1.55 kB | 1.55 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.96 kB | 7.96 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-d1cfb87..dule.js gzip | 1.49 kB | 1.49 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.59 kB | 5.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _error.js | 915 kB | 915 kB | ✓ |
| 404.html | 4.73 kB | 4.73 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 915 kB | 915 kB | ✓ |
| link.js | 964 kB | 964 kB | ✓ |
| routerDirect.js | 957 kB | 957 kB | ✓ |
| withRouter.js | 957 kB | 957 kB | ✓ |
| Overall change | 4.72 MB | 4.72 MB | ✓ |

</details>

</details>
Commit: 3332cc234dd451a5444da3172fe66441a685afbe

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| buildDuration | 13.4s | 13.5s | ⚠️ +48ms |
| nodeModulesSize | 84.9 MB | 84.9 MB | ⚠️ +1.51 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.418 | 2.468 | ⚠️ +0.05 |
| / avg req/sec | 1033.96 | 1013.07 | ⚠️ -20.89 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.339 | 1.341 | 0 |
| /error-in-render avg req/sec | 1866.43 | 1864.59 | ⚠️ -1.84 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..7d3b.js gzip | 11.3 kB | 11.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-56c34a4..e157.js gzip | 7.4 kB | 7.4 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.4 kB | 58.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 7.04 kB | 7.04 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-c8c3589..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53.1 kB | 53.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-fca3d..2eb1.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-8a9d104..0e42.js gzip | 1.55 kB | 1.55 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.96 kB | 7.96 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-d1cfb87..dule.js gzip | 1.49 kB | 1.49 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.59 kB | 5.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 995 B | 995 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| buildDuration | 15.4s | 15.3s | -93ms |
| nodeModulesSize | 84.9 MB | 84.9 MB | ⚠️ +1.51 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..7d3b.js gzip | 11.3 kB | 11.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-56c34a4..e157.js gzip | 7.4 kB | 7.4 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.4 kB | 58.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 7.04 kB | 7.04 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-c8c3589..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53.1 kB | 53.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-fca3d..2eb1.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-8a9d104..0e42.js gzip | 1.55 kB | 1.55 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.96 kB | 7.96 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-d1cfb87..dule.js gzip | 1.49 kB | 1.49 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.59 kB | 5.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _error.js | 915 kB | 915 kB | ✓ |
| 404.html | 4.73 kB | 4.73 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 915 kB | 915 kB | ✓ |
| link.js | 964 kB | 964 kB | ✓ |
| routerDirect.js | 957 kB | 957 kB | ✓ |
| withRouter.js | 957 kB | 957 kB | ✓ |
| Overall change | 4.72 MB | 4.72 MB | ✓ |

</details>

</details>
Commit: 139f0cb334ccb9a9d323dc98bfb6993bf687cd62

File: packages/next/client/image.tsx [hunk 1]
```diff
@@ -493,6 +494,21 @@ function cloudinaryLoader({ root, src, width, quality }: LoaderProps): string {
   return `${root}${paramsString}${normalizeSrc(src)}`
 }
 
+function aliyunLoader({ root, src, width, quality }: LoaderProps): string {
+  // Demo: http://image-demo.oss-cn-hangzhou.aliyuncs.com/example.jpg?x-oss-process=image/resize,w_300/quality,q_90
+  const params = ['image']
+  if (width) {
+    params.push(`resice,w_${width}`)
```
Comment (Jensderond) [inline packages/next/client/image.tsx:501]:
```suggestion
    params.push(`resize,w_${width}`)
```

**Review by Jensderond — CHANGES_REQUESTED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| buildDuration | 12.4s | 12.7s | ⚠️ +312ms |
| nodeModulesSize | 85 MB | 85 MB | ⚠️ +1.51 kB |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.392 | 2.345 | -0.05 |
| / avg req/sec | 1045.24 | 1065.91 | +20.67 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.249 | 1.29 | ⚠️ +0.04 |
| /error-in-render avg req/sec | 2002.3 | 1938.29 | ⚠️ -64.01 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..7d3b.js gzip | 11.3 kB | 11.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-d20d8bb..4991.js gzip | 7.4 kB | 7.4 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.4 kB | 58.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 7.04 kB | 7.04 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-6468b39..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53.1 kB | 53.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-fca3d..2eb1.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-8a9d104..0e42.js gzip | 1.55 kB | 1.55 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.96 kB | 7.96 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-d1cfb87..dule.js gzip | 1.49 kB | 1.49 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.59 kB | 5.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| index.html gzip | 1 kB | 1 kB | ✓ |
| link.html gzip | 1.01 kB | 1.01 kB | ✓ |
| withRouter.html gzip | 998 B | 998 B | ✓ |
| Overall change | 3.01 kB | 3.01 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| buildDuration | 15.1s | 14.4s | -692ms |
| nodeModulesSize | 85 MB | 85 MB | ⚠️ +1.51 kB |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..7d3b.js gzip | 11.3 kB | 11.3 kB | ✓ |
| framework.HASH.js gzip | 39 kB | 39 kB | ✓ |
| main-d20d8bb..4991.js gzip | 7.4 kB | 7.4 kB | ✓ |
| webpack-e067..f178.js gzip | 751 B | 751 B | ✓ |
| Overall change | 58.4 kB | 58.4 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| 677f882d2ed8..dule.js gzip | 7.04 kB | 7.04 kB | ✓ |
| framework.HA..dule.js gzip | 39 kB | 39 kB | ✓ |
| main-6468b39..dule.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-07c5..dule.js gzip | 751 B | 751 B | ✓ |
| Overall change | 53.1 kB | 53.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| polyfills-4b..e242.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-7231d4b..5856.js gzip | 1.28 kB | 1.28 kB | ✓ |
| _error-fca3d..2eb1.js gzip | 3.44 kB | 3.44 kB | ✓ |
| hooks-d4591d..e7c2.js gzip | 887 B | 887 B | ✓ |
| index-17468f..5d83.js gzip | 227 B | 227 B | ✓ |
| link-8a9d104..0e42.js gzip | 1.55 kB | 1.55 kB | ✓ |
| routerDirect..924c.js gzip | 284 B | 284 B | ✓ |
| withRouter-7..c13d.js gzip | 284 B | 284 B | ✓ |
| Overall change | 7.96 kB | 7.96 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _app-75d3a82..dule.js gzip | 625 B | 625 B | ✓ |
| _error-4469a..dule.js gzip | 2.29 kB | 2.29 kB | ✓ |
| hooks-cbf13f..dule.js gzip | 387 B | 387 B | ✓ |
| index-b9a643..dule.js gzip | 226 B | 226 B | ✓ |
| link-d1cfb87..dule.js gzip | 1.49 kB | 1.49 kB | ✓ |
| routerDirect..dule.js gzip | 284 B | 284 B | ✓ |
| withRouter-f..dule.js gzip | 282 B | 282 B | ✓ |
| Overall change | 5.59 kB | 5.59 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _buildManifest.js gzip | 322 B | 322 B | ✓ |
| _buildManife..dule.js gzip | 330 B | 330 B | ✓ |
| Overall change | 652 B | 652 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | vercel/next.js canary  | Gerhut/next.js image-loaders/aliyun-oss | Change |
| - | - | - | - |
| _error.js | 915 kB | 915 kB | ✓ |
| 404.html | 4.73 kB | 4.73 kB | ✓ |
| hooks.html | 3.92 kB | 3.92 kB | ✓ |
| index.js | 916 kB | 916 kB | ✓ |
| link.js | 965 kB | 965 kB | ✓ |
| routerDirect.js | 958 kB | 958 kB | ✓ |
| withRouter.js | 958 kB | 958 kB | ✓ |
| Overall change | 4.72 MB | 4.72 MB | ✓ |

</details>

</details>
Commit: 1cffa519dc04b9c2fd43eb824e7bb555c88e70e0

**Comment by Timer** [comment]
Hi! We just added the ability to pass your own custom loader to the Image component, so we're no longer accepting new built-in providers.

See this PR to learn how to use it: #20788

Thanks!

**Pull request closed without merging**